system/Autoloader/FileLocator.php

```diff
@@ -18,15 +18,11 @@
 class FileLocator
 {
     /**
-     * The Autoloader to use.
-     *
-     * @var Autoloader
+     * @param Autoloader $autoloader The Autoloader to use.
      */
-    protected $autoloader;
-
-    public function __construct(Autoloader $autoloader)
-    {
-        $this->autoloader = $autoloader;
+    public function __construct(
+        protected Autoloader $autoloader
+    ) {
```

Comment:
For example, if a dev extends `FileLocator` and it has `protected $autoloader;`,
this change is a breaking change.
https://3v4l.org/NhqDb

Reply:
There are possible options:

- make a note about property re-definition possible breaking change without add type
- wait to push to 4.4 only

Reply:
Another way;: make Rector extension to help user migrating to new definition.

Reply:
Removing type in the constructor makes no errors (I don't know why), 
https://3v4l.org/2A85L
but I don't want to remove already declared types in constructors.

Reply:
Better way: make a patch to rector to:

- skip non-typed property for protected/public property, except define `INLINE_PUBLIC` config that allow bc break on non-final class, same like TypedPropertyRector way

That's just quick guessing, probably there is a better way.

Reply:
@kenjis here possibly the solution, I create PR to make it configurable `INLINE_PUBLIC` default false:

- https://github.com/rectorphp/rector-src/pull/3126

Reply:
This seems the best approach to me. Glad your patch was already accepted!